hashtables/ex1/ex1.py

```diff
@@ -13,8 +13,33 @@ def get_indices_of_item_weights(weights, length, limit):
     YOUR CODE HERE
     """
 
+    # loop through the weights array (using the length variable to 
+    # create the duration of the loop)
+        # insert values within the hashtable 
+        # keys will be weights[index] - limit 
+        # values will be index 
+
+    # create and additional for loop that will loop through the 
+    # weights array using again length as the duration 
+        # if retrieve(hashtable(weights[index])) does not return None 
+            # return both index and the value within the hashtable
+
+    for index in range(length):
+        hash_table_insert(ht, (limit-weights[index]), index)
+
+    for index_i in range(length):
+        if hash_table_retrieve(ht, (weights[index_i])):
+            index_j = hash_table_retrieve(ht, (weights[index_i]))
+            if index_j > index_i:
+                return [index_j, index_i]
+            else:
+                return [index_i, index_j]
```

Comment:
I think this solution is fine (and it was also my first pass solution when I took this SC) but something to consider when reflecting over this code is we are always moving towards the higher index while iterating - so technically we could always return the value we are currently iterating over first which is guaranteed to always be the higher valued index, and our previous item that we already stored in the hash table, which will always be our lower valued index.